File: codility/EquiLeader.js
function solution(A) {
    // Implement your solution here
    const obj = {};
    const half = A.length / 2;
    let halfKey;
    for (let i = 0; i < A.length; i++) {
        const v = A[i];
        if (obj[v]) {
            obj[v]++;
        } else {
            obj[v] = 1;
        }
        if (obj[v] > half) {
            halfKey = v;
        }
    }
    if (typeof halfKey === "undefined") {
        return 0;
    }
    const halfCount = obj[halfKey];
    let result = 0,
        leftCount = 0;
    for (let i = 0; i < A.length; i++) {
        const v = A[i];
        if (halfKey === v) {
            leftCount++;
        }
        if (
            leftCount > (i + 1) / 2 &&
            halfCount - leftCount > (A.length - i - 1) / 2
        ) {
            result++;
        }
    }
    return result;
}

File: codility/python/EquiLeader.py
def solution(A):
    # Implement your solution here
    n = len(A)
    dict = {}
    leader_num = None
    for i in range(n):
        a = A[i]
        if dict.get(a, None) == None:
            dict[a] = {'count':1, 'list': [i]}
        else:
            dict[a]['count'] += 1
            dict[a]['list'].append(i)
            if dict[a]['count'] > n/2:
                leader_num = a
    if leader_num == None:
        return 0
    list = dict[leader_num]['list']
    index = 0
    count_1 = 0
    count = 0
    m = len(list)
    for i in range(n):
        if i == list[index]:
            if index < m -1 :
                index += 1
            count_1 += 1
        if count_1 > (i+1)/2 and m - count_1 > (n-i-1)/2:
            count += 1
    return count